### PR TITLE
Fim delegations fix

### PR DESCRIPTION
**Pull request by ibaldin**
Updates to reflect the change in delegations property format in FIM.

### CODE REVIEW

**Comment by kthare10** [comment]
Changes to incorporate the new delegation format are needed in network_node_control.py as well.

**Comment by kthare10** [comment]
network_node_inventory.py, line 243
```
if graph_node.capacity_delegations is None or len(graph_node.capacity_delegations) < 1 or reservation is None:
```

It currently checks for len of delegations i.e.  len(graph_node.capacity_delegations) which does not apply anymore and can be removed

**Comment by kthare10** [comment]
fim_helper.py: update_node function which updates the graph node in ASM throws an exception if label_allocations are none due to the new assert statement.

```
   File "/usr/src/app/fabric_cf/actor/core/kernel/slice.py", line 292, in update_slice_graph
     FimHelper.update_node(graph_id=self.graph_id, sliver=sliver)
   File "/usr/src/app/fabric_cf/actor/fim/fim_helper.py", line 200, in update_node
     node.components[cname].set_properties(label_allocations=component.label_allocations,
   File "/usr/local/lib/python3.9/site-packages/fim/user/component.py", line 138, in set_properties
     comp_sliver.set_properties(**kwargs)
   File "/usr/local/lib/python3.9/site-packages/fim/slivers/base_sliver.py", line 159, in set_properties
     self.__getattribute__('set_' + k)(v)
   File "/usr/local/lib/python3.9/site-packages/fim/slivers/base_sliver.py", line 104, in set_label_allocations
     assert(lab is None or isinstance(lab, Labels))
 AssertionError
```

**Comment by ibaldin** [comment]
I fixed the problem in `network_node_inventory.py`, however the other problem is a bit of a mystery. You can see the assert statement allows for label to be None (or it has to be an instance of Labels). I could not fully trace the code where labels in the sliver are assigned, but something is happening there where these allocated labels aren't actually a Labels object.

File: fabric_cf/actor/core/policy/network_node_inventory.py [hunk 1]
```diff
@@ -121,29 +117,28 @@ def __check_component_labels_and_capacities(self, *, available_component: Compon
...
-            for delegated in delegation_list:
-                if Pool.ispoolmention(delegated) or Pool.ispooldefinition(delegated):
-                    # ignore pool mentions and definitions for now
-                    continue
-                self.logger.debug(f"available_label_delegations : {delegated} {type(delegated)} for component "
-                                  f"{available_component}")
-                delegated_label = Labels().set_fields(**delegated)
-                requested_component.label_allocations = delegated_label
-                break
+        delegation_id, deleg = capacity_delegations.get_sole_delegation()
```
Comment (kthare10) [inline fabric_cf/actor/core/policy/network_node_inventory.py:134]: fabric_cf/actor/core/policy/network_node_inventory.py: Line 134
This should be changed to label_delegations.get_sole_delegation(). I think this is causing the Assertion failure in UpdateNode when updating ASM in FimHelper.

Comment (ibaldin) [reply fabric_cf/actor/core/policy/network_node_inventory.py:134]: Oh yeah. D'oh. Must be cut-and-paste artifact of some sort.

**Review by kthare10 — APPROVED** [review]

**Pull request merged**